src/adtypes_interface.jl

```diff
@@ -172,18 +172,35 @@ for detector in (:TracerSparsityDetector, :TracerLocalSparsityDetector)
 end
 
 ## Stable API to allow packages like DI to allocate caches of tracers
+
+"""
+    jacobian_tracer_type(x, detector)
+
+Return the tracer type used for Jacobian sparsity detection.
+"""
+jacobian_tracer_type(x, ::TracerSparsityDetector{TG}) where {TG} = TG
+jacobian_tracer_type(x, ::TracerLocalSparsityDetector{TG}) where {TG} = Dual{eltype(x),TG}
+
+"""
+    hessian_tracer_type(x, detector)
+
+Return the tracer type used for Hessian sparsity detection.
+"""
+hessian_tracer_type(x, ::TracerSparsityDetector{TG,TH}) where {TG,TH} = TH
+function hessian_tracer_type(x, ::TracerLocalSparsityDetector{TG,TH}) where {TG,TH}
+    return Dual{eltype(x),TH}
+end
+
```

Comment:
Adding `x::Real` might make sense here.

Duals expect `Dual{P<:Real,T<:AbstractTracer}` and we probably also don't want to return a global tracer when passed an array.

Reply:
Solved, assuming you meant `AbstractArray{<:Real}`